packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/05-pending-call.mdx

```diff
@@ -0,0 +1,285 @@
+
+## Project setup and prerequisites
+
+Make sure you have the following prerequisites checked:
+
+1. [Registered Stream account](https://getstream.io/try-for-free/)
+2. Have an app created in the [Stream's dashboard](https://dashboard.getstream.io/) to obtain app API key and secret.
+3. Initiate the project (you can follow our [introductory tutorial setup guide](../../tutorials/video-calling#set-up-for-success))
+4. Have installed the Stream video and chat SDKs in the project:
+
+```shell
+npm install @stream-io/video-react-sdk stream-chat-react stream-chat
+```
+
+```shell
+yarn add @stream-io/video-react-sdk stream-chat-react stream-chat
+```
+
+## App boilerplate
+In order we can start performing calls we need a `StreamVideoClient` instance connected to the Stream's server network with app credentials. For more information on how to do that, please take a look at our [authentication guide](../../guides/client-auth#setting-up-a-streamvideoclient-and-connecting-a-user).
+
+## Pending call panel implementation
+The component will rely on data accessed via our utility hooks:
+
+- [`useCall`](../../call-engine/hooks-and-contexts#usecall)
+- [`useCallCallingState`](../../call-engine/hooks-and-contexts#usecallcallingstate)
+- [`useCallMetadata`](../../call-engine/hooks-and-contexts#usecallmetadata)
+- [`useCallMembers`](../../call-engine/hooks-and-contexts#usecallmembers)
+- [`useMediaDevices`](../../call-engine/hooks-and-contexts#usemediadevices)
+
+### User preview
+
+The panel will render [`VideoPreview`](todo docs link) so that the user can see the camera input or avatars if video preview is disabled. The avatar display logic is as follows:
+- show all the called users' avatars if the call is outgoing (who am I calling)
+- show the avatar of the person who initiated the call in case of an incoming call (who is calling me)
+
+:::note
+Note, how the initial video state is handled by [`MediaDevicesProvider`](../../call-engine/hooks-and-contexts#mediadevicesprovider). We only consume the API through corresponding context consumer hook [`useMediaDevices`](../../call-engine/hooks-and-contexts#usemediadevices).
+:::
+```tsx
+import {
+  useCall,
+  useCallMembers,
+  useCallMetadata,
+  useMediaDevices,
+} from '@stream-io/video-react-sdk';
+
+type PendingCallPanelProps = {
+  showMemberCount?: number;
+};
+
+export const CustomPendingCallPanel = ({
+  showMemberCount = 3,
+}: PendingCallPanelProps) => {
+  const call = useCall();
+  const metadata = useCallMetadata();
+  const members = useCallMembers();
+  // highlight-next-line
+  const { initialVideoState } = useMediaDevices();
+
+  if (!call) return null;
+
+  const caller = metadata?.created_by;
+  // show the caller if this is an incoming call or show all the users I am calling to
+  const membersToShow = call.isCreatedByMe
+    ? members
+        ?.slice(0, showMemberCount)
+        .map(({ user }) => user)
+        .filter((u) => !!u) || []
+    : caller
+    ? [caller]
+    : [];
```

Comment:
This snippet is a bit hard to be followed by a reader. Can you break it down into a few `if-else` branches?